API/SelectU.Core/Services/ReviewService.cs

```diff
@@ -0,0 +1,94 @@
+        {
+            _unitOfWork = context;
+        }
+
+
+        public async Task CreateScholarshipApplicationRatingAsync(UserRatingDTO userRatingDTO)
+        {
+            var scholarshipApplication = await _unitOfWork.ScholarshipApplications.GetAsync(userRatingDTO.ScholarshipApplicationId);
+
+            if (scholarshipApplication == null) throw new ReviewException($"Unable able to add rating as the application does not exist");
+
+            if (scholarshipApplication.Ratings.FirstOrDefault(x => x.ReviewerId == userRatingDTO.ReviewerId) == null) throw new ReviewException($"Unable able to create rating as the reviewer has an existing review");
+
+            UserRating userRating = new UserRating()
+            {
+                ScholarshipApplicationId = userRatingDTO.ScholarshipApplicationId,
+                Rating = userRatingDTO.Rating,
+                Id = userRatingDTO.Id,
+                ReviewerId = userRatingDTO.ReviewerId,
+                ApplicantId = userRatingDTO.ApplicantId,
+                Comment = userRatingDTO.Comment
+            };
+
+            scholarshipApplication.Ratings.Add(userRating);
+            _unitOfWork.ScholarshipApplications.Update(scholarshipApplication);
+
+            await _unitOfWork.CommitAsync();
+
+        }
+
+        public async Task UpdateScholarshipApplicationRatingAsync(UserRatingDTO userRatingDTO, bool isAdmin)
+        {
+            var scholarshipApplication = await _unitOfWork.ScholarshipApplications.GetAsync(userRatingDTO.ScholarshipApplicationId);
+
+            if (scholarshipApplication == null)
+            {
+                throw new ReviewException($"Unable able to add rating as the application does not exist");
+            }
+
+            UserRating userRating = new UserRating()
+            {
+                ScholarshipApplicationId = userRatingDTO.ScholarshipApplicationId,
+                Rating = userRatingDTO.Rating,
+                Id = userRatingDTO.Id,
+                ReviewerId = userRatingDTO.ReviewerId,
+                ApplicantId = userRatingDTO.ApplicantId,
+                Comment = userRatingDTO.Comment
+            };
+            var originalRating = scholarshipApplication.Ratings.FirstOrDefault(x => x.Id == userRating.Id && (x.ReviewerId == userRatingDTO.ReviewerId || isAdmin));
+            if (originalRating == null) throw new ReviewException($"Unable able to update rating as you need to be the owner of the review");
+            scholarshipApplication.Ratings.Remove(originalRating);
+            scholarshipApplication.Ratings.Add(userRating);
+            _unitOfWork.ScholarshipApplications.Update(scholarshipApplication);
+
+            await _unitOfWork.CommitAsync();
+
+
+        }
+
+        public async Task DeleteScholarshipApplicationRatingAsync(Guid RatingId, string CreatorId, bool isAdmin)
+        {
+            var userRating = await _unitOfWork.UserRating.GetAsync(RatingId.ToString());
```

Comment:
dont convert to string, this will throw an exception

Reply:
requires string

Reply:
nvm